### ISSUE TITLE
Plugin leaks memory

### ISSUE THREAD

**Issue by ened** [miguelpruivo/flutter_file_picker#206]
**Describe the bug**
Running the plugin in an App that has LeakCanary enabled, the following log will appear:

```
Analysis in progress, working on: PARSING_HEAP_DUMP
Analysis in progress, working on: EXTRACTING_METADATA
Analysis in progress, working on: FINDING_LEAKING_INSTANCES
Analysis in progress, working on: FINDING_PATHS_TO_LEAKING_OBJECTS
Analysis in progress, working on: FINDING_DOMINATORS
Analysis in progress, working on: COMPUTING_NATIVE_RETAINED_SIZE
Analysis in progress, working on: COMPUTING_RETAINED_SIZE
Analysis in progress, working on: BUILDING_LEAK_TRACES
Analysis in progress, working on: REPORTING_HEAP_ANALYSIS
====================================
HEAP ANALYSIS RESULT
====================================
1 APPLICATION LEAKS
References underlined with "~~~" are likely causes.
Learn more at https://squ.re/leaks.
22899 bytes retained
┬
├─ android.net.ConnectivityThread
│    Leaking: NO (PathClassLoader↓ is not leaking)
│    Thread name: 'ConnectivityThread'
│    GC Root: Local variable in native code
│    ↓ thread ConnectivityThread.contextClassLoader
├─ dalvik.system.PathClassLoader
│    Leaking: NO (Object[]↓ is not leaking and A ClassLoader is never leaking)
│    ↓ PathClassLoader.runtimeInternalObjects
├─ java.lang.Object[]
│    Leaking: NO (FilePickerPlugin↓ is not leaking)
│    ↓ array Object[].[946]
├─ com.mr.flutter.plugin.filepicker.FilePickerPlugin
│    Leaking: NO (a class is never leaking)
│    ↓ static FilePickerPlugin.instance
│                              ~~~~~~~~
├─ io.flutter.app.FlutterPluginRegistry$FlutterRegistrar
│    Leaking: UNKNOWN
│    ↓ FlutterPluginRegistry$FlutterRegistrar.this$0
│                                             ~~~~~~
├─ io.flutter.app.FlutterPluginRegistry
│    Leaking: UNKNOWN
│    ↓ FlutterPluginRegistry.mActivity
│                            ~~~~~~~~~
╰→ com.company.app.MainActivity
​     Leaking: YES (Activity#mDestroyed is true and ObjectWatcher was watching this)
​     key = 01f51324-3411-4776-ace1-ad77cf2cb72f
​     watchDurationMillis = 5160
​     retainedDurationMillis = 158
====================================
0 LIBRARY LEAKS
Leaks coming from the Android Framework or Google libraries.
====================================
METADATA
Please include this in bug reports and Stack Overflow questions.
Build.VERSION.SDK_INT: 29
Build.MANUFACTURER: samsung
LeakCanary version: 2.0-beta-4
```

This is due to the plugin keeping a static reference to the instance, probably for easier handling of activity results. 

**Issue details**
1. Platform (Android/iOS):

Android

2. OS version

10

**Flutter Version details**

```
❯ flutter doctor -v
[✓] Flutter (Channel beta, v1.14.6, on Mac OS X 10.15.3 19D76, locale en-GB)
    • Flutter version 1.14.6 at /Users/sebastian/dev/flutter
    • Framework revision fabeb2a16f (3 weeks ago), 2020-01-28 07:56:51 -0800
    • Engine revision c4229bfbba
    • Dart version 2.8.0 (build 2.8.0-dev.5.0 fc3af737c7)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/sebastian/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = /Users/sebastian/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 43.0.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.42.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.8.1

[✓] Connected device (1 available)
    • SM G975F • R58M36QGTJM • android-arm64 • Android 10 (API 29)

• No issues found!
```

I'll try to work on a PR this week, this may also be a good time for general refactoring.

**Comment by miguelpruivo** [comment]
I'm not sure if this is going to be worth right now @ened as I'm already working in a refactor on Android side to make it compatible to Android add 2 app.   

Anyway, thank you. I'll leave this open for now.

**Comment by ened** [comment]
Hi @miguelpruivo do you work on this in a private branch or something public? I'd be happy to review a WIP branch and check for the issue myself.

**Comment by miguelpruivo** [comment]
@ened it's not public yet. But I'm adding [V2 support](https://flutter.dev/docs/development/packages-and-plugins/plugin-api-migration) which I believe that might remove that leak you are pointing out. 

I'll keep you updated and tell you when the branch is up on origin.

Thank you!

**Comment by ened** [comment]
@miguelpruivo how is the V2 support doing? Do you need help / will you accept a PR?

**Comment by miguelpruivo** [comment]
@ened it's a work in progress. Currently I'm also refactoring some things along the way, just hand't the time yet to finish it. Do you have something already set up?

**Comment by ened** [comment]
It should be straightforward. Perhaps upload what you have to a develop branch and I can take a look there.

**Comment by miguelpruivo** [comment]
@ened thank you, it's almost done, but I'm also refactoring some things along the way, that's why it took a bit more. :) I have plans to have it published by this week. Right now I'm working only on a local branch.

Yesterday I posted a comment regarding this on #195, but this two issues should be in sync with each other.

**Comment by miguelpruivo** [comment]
V2 embedding support was added and published with [1.5.0](https://pub.dev/packages/file_picker#150).

You should haven't this issue by now.

As a side note, have in mind, that along with it, now FileType has been refactored to match camelCase enum Dart guideline, so you'll need to refactor your code to match accordingly (eg. FileType.ALL now is FileType.all).

Thank you for being patient. 🎉
